src/backend/core/api/viewsets.py

```diff
@@ -322,22 +326,35 @@ def get_serializer_context(self):
         context["team_id"] = self.kwargs["team_id"]
         return context
 
+    def get_serializer_class(self):
+        if self.action in ["list", "retrieve"]:
+            return self.list_serializer_class
+        return self.detail_serializer_class
+
     def get_queryset(self):
         """Return the queryset according to the action."""
         queryset = super().get_queryset()
         queryset = queryset.filter(team=self.kwargs["team_id"])
 
-        if self.action == "list":
+        if self.action in ["list", "retrieve"]:
```

Comment:
very-nit: Let's use the right data structure whenever it's possible. Here we could use a set. It's a good habit, that slightly improve performances in some situations.

```suggestion
        if self.action in {"list", "retrieve"}:
```